[50109] Breadcrumb and menu structure is inconsistent for user administration

HDinger opened the pull request:
In most of our Administration modules, the breadcrumb does not follow the structure we decided to use since we introduced the PageHeaders. So this PR goes ahead, and introduced the `Primer::OpenProject::Pageheader` component with the correct breadcrumbs in the module Administration/User settings.

**Todo**

Introduce PageHeader component in:
- [x] Admin/User settings
- [x] Admin/Users
  - [x] Index
  - [x] Show
  - [x] Edit
    - [x] replace the tab rendering with the PageHeader tabNav?
- [x] Placeholder users
  - [x] Index
  - [x] Show
  - [x] Edit
    - [x] replace the tab rendering with the PageHeader tabNav?
- [x] Groups
  - [x] Index
  - [x] Show
  - [x] Edit
    - [x] replace the tab rendering with the PageHeader tabNav?
- [x] Roles and permissions
  - [x] Index
  - [x] Show
  - [x] Edit
- [x] Permissions report
- [x] Avatars

Because Avatars uses a generic "plugin" page, other modules are affected as well that use the same page

- [x] Time & Costs
  - [x] Settings
  - [x] Cost types

https://community.openproject.org/projects/openproject/work_packages/50109/activity

HDinger committed:
Introduce PageHeader component in Administration / User settings

HDinger committed:
Introduce PageHeader in Administration / Users

HDinger committed:
Introduce PageHeader in Administration / Roles

HDinger committed:
Use tabNav element of the pageHeader in Admin -> Users -> Edit user

HDinger committed:
Introduce PageHeader component in Administration -> user settings -> Placeholder users

HDinger committed:
Use helper instead of repeating code

HDinger committed:
Introduce PageHeader component in Administration -> User settings -> Groups

HDinger committed:
Introduce PageHeader component in plugin settings pages (e.g avatar or cost settings)

HDinger committed:
Introduce PageHeader component in Time & Costs -> Cost types to harmonise it with the Time & Costs -> Settings page

HDinger committed:
Fix user_helper

ulferts commented on app/helpers/users_helper.rb:
```diff
@@ -87,41 +87,51 @@ def full_user_status(user, include_num_failed_logins = false)
   def build_change_user_status_action(user)
+    result = "".html_safe
+    iterate_user_statusses(user) do |title, name|
+      result << ((yield title, name) + " ".html_safe)
```
The `.html_safe` on the empty string " " should not be necessary. If you concatenate to an `html_safe?` string, the concatenated string will be escaped automatically so that the result is safe again.

```
[1] pry(main)> x = ''.html_safe
=> ""
[2] pry(main)> x.html_safe?
=> true
[3] pry(main)> x << " "
=> " "
[4] pry(main)> x.html_safe?
=> true
```

ulferts commented on app/views/placeholder_users/_toolbar.html.erb:
```diff
@@ -26,14 +26,51 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+        end
+      end
+    end if tabs.present?
+  end
```
This is currently a partial. In that partial, some variables are defined in order to avoid recalculating them. I guess the better approach would be to turn the toolbar into a component. 

On a few pages we have a `*_page_header_component`. The question now is, whether we want to establish that as a pattern.

ulferts replied:
Taking a look at other places where the block is added directly in the `show`/`edit`/... page, the header block now sometimes takes a lot of lines compared to the rest of the content. That in my opinion lends weight to the argument of wanting to put those parts into components.

ulferts commented on app/views/users/_toolbar.html.erb:
```diff
@@ -26,42 +26,78 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+    end
+
+    if current_user.admin?
+      unless current_user.id == @user.id
```
💡 Since we do this over and over again in the code, we could also introduce a `@user.current?` method.

ulferts commented on app/views/users/index.html.erb:
```diff
@@ -38,21 +38,38 @@ See COPYRIGHT and LICENSE files for more details.
 <% users_info = user_limit && content_tag(:div) do %>
```
I know this existed before but this is plain weird to read. It also leads to later on having two variables referenced. This could be simplified:

```
<% if user_limit %>
  <div>
     <%= t(:label_enterprise_active_users, current: ...
  </div>
<% end %>

HDinger committed:
Adapt tests to new PageHeader structure

ulferts reviewed (approved):
The change looks fine, good work.

While working through the code I came up with small remarks and ideas that however don't need to be fixed before merging.

HDinger commented:
Hi @ulferts 

I addressed your remarks and extracted the code for the PageHeaders into separate components in a separate PR: https://github.com/opf/openproject/pull/16151 . I will merge this PR then.
